tcms/testcases/static/testcases/js/mutable.js

```diff
@@ -0,0 +1,37 @@
+$(document).ready((function () {
+    populateProductCategory();
+
+    if ($('#id_category').val() === null || !$('#id_category').val().length) {
+        update_category_select_from_product();
+    }
+
+    $('#add_id_product').click(function () {
+        return showRelatedObjectPopup(this);
+    });
+
+    $('#add_id_category').click(function () {
+        return showRelatedObjectPopup(this);
+    });
+
+    document.getElementById('id_product').onchange = function () {
+        $('#id_product').selectpicker('refresh');
+        populateProductCategory();
+    };
+
+    document.getElementById('id_category').onchange = function () {
+        $('#id_category').selectpicker('refresh');
+    };
+
+    $('.selectpicker').selectpicker();
+    $('.bootstrap-switch').bootstrapSwitch();
+}));
+
+function populateProductCategory() {
+    let href = $('#add_id_category')[0].href;
+    $('#add_id_category')[0].href = href.slice(0, href.indexOf('&product'));
+    $('#add_id_category')[0].href += `&product=${$('#id_product').val()}`;
+
+    $('#add_id_category').find('option').remove();
```

Comment:
Now this selector is wrong and this is why refreshing categories doesn't remove the previous entries.